src/serde/deserialize_program.rs

```diff
@@ -188,10 +188,7 @@ impl<'de> de::Visitor<'de> for ValueAddressVisitor {
             None
         };
         let offset = if let Some((_, offset_str)) = regex_captures!(r"(-?\d+)", value) {
-            match i32::from_str(offset_str) {
-                Ok(offset) => offset,
-                Err(e) => return Err(e).map_err(de::Error::custom),
-            }
+            i32::from_str(offset_str).unwrap_or(-1)
```

Comment:
Why `-1` when it doesn't match? Wouldn't `0` make more sense (no offset -> no addition -> addition of `0`)?

Reply:
This is an intermediate solution, as this case is not currently handled and produces failed parse errors on references that don't belong to hints. For example: "cast(18446744073709551616, felt)" is not contemplated. Any offset value would be wrong as there is no register. I will change it to 0 for consistency so that it is None + 0 = None.